single-function/valid-parentheses/submission.js

```diff
@@ -0,0 +1,48 @@
+const isValidParentheses = (str) => {
+  if(str.length <= 1) { return false }
+  const parenMap = { '{':'}', '(':')', '[':']' };
+  const stack = [];
+  for (let i = 0; i < str.length; i++) {
+    if (parenMap[str[i]]) {
+      stack.push(parenMap[str[i]]);
+    } else if (stack.pop() !== str[i]) {
+      return false;
+    }
+  }
+  return (str.length % 2 !== 0) || (stack.length !== 0) || true;
+};
+
+/*
+Input: '()()()'
+Output: boolean
+Constraints: n/a
+
+Strategy: Step through the given string, if a valid opening paren push closing paren on to stack.
+If a non valid opening paren found pop from the stack and not equal to the current string value invalidate string.
+
+Transformation Steps:
+  Acceptable Chars: {}[]()
+  Given '(())'
+  Step 1: '(' found push ')' onto stack
+  Step 2: '(' found push ')' onto stack
+  Step 3: ')' not found, popped value from stack matches
```

Comment:
Hm, I think it'd be better if we did the slightly more realistic version of this problem, which is to make sure that parens balance _within a piece of real code_.  That way people won't get tripped up if they do see that common version of this problem.  I.e.,  a bunch of alphanumeric characters also would not be found, which makes this solution not quite work for that expanded spec.

See
https://www.geeksforgeeks.org/check-for-balanced-parentheses-in-an-expression/